manifests/server.pp

```diff
@@ -131,6 +135,7 @@
   validate_string($rootdn)
   validate_string($rootpw)
   validate_absolute_path($directory)
+  validate_string($backend)
```

Comment:
Unfortunately not, logic is applied on the master, not the client. An array of supported backends would be enough I think, given it would catch people's typos, etc.